src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java

```diff
@@ -140,23 +143,33 @@ public void accept(CtVisitor visitor) {
 		return findClass();
 	}
 
+	private static Map<String, Class> classByQName = Collections.synchronizedMap(new HashMap<>());
```

Comment:
static field = global state = evil

is that required? is there an alternative?

Reply:
> alternative

an LRUMap identity map with key = ClassLoader and value  = Map<ClassName, Class>

Reply:
Both solutions will improve performance and break nothing so I am OK to merge it.